src/test/java/org/jabref/logic/importer/fileformat/BibtexParserTest.java

```diff
@@ -857,6 +857,15 @@ void parseRecognizesPreambleWithConcatenation() throws IOException {
         assertEquals(Optional.of("\"some text\" # \"and \\latex\""), result.getDatabase().getPreamble());
     }
 
+    @Test
+    void parseRecognizesPreambleWithArbitraryBracketPairAndSpace() throws IOException {
+        ParserResult result = parser
+                .parse(new StringReader("@preamble{some (  {text    )}) and \\latex}"));
+        //                                                               |> the rest got cut-off
```

Comment:
I do not understand "got cut-off". Can you elaborate? Maybe, it is just not required, beucause the preamble itself is tested.

Reply:
so in current implementation, as long as jabref get enough of pairing brackets, in this case `{ ( {  ) } )`, it returns immediately and ignores the rest of the input.